scripts/opencl/opencl.go

```diff
@@ -36,28 +36,28 @@ func compileSources() (string, error) {
 	dir := filepath.Join(filepath.Dir(filename), sourceDir)
 	files, err := ioutil.ReadDir(dir)
 	if err != nil {
-		return "", errors.Errorf("couldn't read directory %s: %s", dir, err.Error())
+		return "", errors.Wrapf(err, "read directory: %s", dir)
 	}
 
 	contents := make(map[string][]byte)
 	for _, file := range files {
 		filepath := filepath.Join(dir, file.Name())
 		data, err := ioutil.ReadFile(filepath)
 		if err != nil {
-			return "", errors.Errorf("couldn't read file %s: %s", filepath, err.Error())
+			return "", errors.Wrapf(err, "read file: %s", filepath)
 		}
 		contents[file.Name()] = data
 	}
 	var out strings.Builder
 	if _, ok := contents[headerFile]; !ok {
-		return "", errors.Errorf("missing %s file", headerFile)
+		return "", errors.Wrapf(err, "missing header: %s file", headerFile)
```

Comment:
The question still stands. Where does this `err` variable come from?

Reply:
ah got it, when there is no variable defined `err` in the function I can use `errors.ErrorF`

another thing i changed some err.Error () to errors lib is it prudent to do this?